docs/src/dev-docs/adding-models/interface.rst

```diff
@@ -0,0 +1,55 @@
+Interface
+#########
+
+In order to add a new model, two things are required:
+
+- a ``Model`` class
+- a ``train_model`` function
```

Comment:
I think it's `train()` function instead of `train_model`. I would also add a few lines on where these files should be located.